lib/database/mongodb.ts

```diff
@@ -29,15 +30,23 @@ export default (mPool: MongoClient): MongoDbService => {
     getWord: (wordId: ObjectId) =>
       getWordsCollection().findOne({ _id: new ObjectId(wordId) }),
 
-    getWords: async ({ page, itemsPerPage }: PaginatedWordsRequest) => {
-      const getPaginatedPromise = getWordsCollection()
-        .find()
+    getWords: async ({ page, itemsPerPage, tags }: PaginatedWordsRequest) => {
+      const getTagsFilter = () => {
+        if (tags === undefined) return {};
+        else if (tags.length === 0)
```

Comment:
The formatting here is a bit funky, also you don't need `else` after `return` 🙂 
```typescript
if (tags === undefined) return {};
if (tags.length === 0) return { $or: [{ tags: undefined }, { tags: [] }] };

return { tags: { $all: [...tags] } };
```
I guess it is a matter of preference.

Reply:
Removed the `else`s.